src/main/webapp/resources/js/components/samples/components/GenomeAssemblyList.jsx

```diff
@@ -30,20 +35,86 @@ export function GenomeAssemblyList({ removeSampleFiles = () => {} }) {
     downloadGenomeAssemblyFile({ sampleId, genomeAssemblyId });
   };
 
+  /*
+  Set default genome assembly for sample to be used for analyses
+   */
+  const updateDefaultGenomeAssembly = (genomeAssembly) => {
+    const { fileInfo: genomeAssemblyObj } = genomeAssembly;
+
+    updateSampleDefaultGenomeAssembly({
+      sampleId: sample.identifier,
+      genomeAssemblyId: genomeAssemblyObj.identifier,
+    })
+      .then(({ data }) => {
+        dispatch(setDefaultGenomeAssembly(genomeAssemblyObj));
```

Comment:
Why do you need to store this as state?  The point to using redux toolkit is that it handles the state and you just need to invalidate to update.  It has it's own cache to handle this.

Reply:
The default tag doesn't update in the UI without it. If you look in samples.js we invalidateTags: ["SampleDetails"]. However even with this it doesn't update the UI side even when a different assembly is updated. Same thing applies to the setting of a default sequencing object

Reply:
I think it is because the `tag` here is set up wrong.  If you look at the docs for [invalidating tags](https://redux-toolkit.js.org/rtk-query/usage/automated-refetching#selectively-invalidating-lists), we do not actually set the tag type on the object itself. I think where we dropped the ball was, instead of:

```javascript
providesTags: ["SampleDetails"],
```

```javascript
providesTags: (result) =>  [{ type: 'SampleDetails', id: details.identifier }],
```

I have not played with it, but it seems more like what they are looking for and probably why we are not getting invalidation and refetching.